packages/flutter_tools/gradle/flutter.gradle

```diff
@@ -317,9 +317,11 @@ class FlutterPlugin implements Plugin<Project> {
             pluginProject.android.buildTypes {
                 "${buildType.name}" {}
             }
-            pluginProject.dependencies.add(
-                "${buildType.name}CompileOnly",
-                "io.flutter:flutter_embedding_$flutterBuildMode:$engineVersion")
+            addApiDependencies(
```

Comment:
nit: does the Closure name `addEmbeddingCompileOnlyDependency` still makes sense?

Reply:
Good catch. I renamed it, but after looking at some test cases, it would need to use `api` or `compileOnly` depending on the Gradle version.